File: infrastructure/src/main/java/com/study/platform/global/auth/adapter/RedisTokenRepositoryAdapter.java
package com.study.platform.global.auth.adapter;

import com.study.platform.global.auth.port.TokenRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Component;

import java.time.Duration;
import java.util.Optional;

@Component
@RequiredArgsConstructor
public class RedisTokenRepositoryAdapter implements TokenRepository {

    private final StringRedisTemplate stringRedisTemplate;

    @Override
    public void save(String key, String value, Duration ttl) {
        stringRedisTemplate.opsForValue().set(key, value, ttl);
    }

    @Override
    public Optional<String> find(String key) {
        return Optional.ofNullable(stringRedisTemplate.opsForValue().get(key));
    }

    @Override
    public void delete(String key) {
        stringRedisTemplate.delete(key);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/global/auth/adapter/RedisTokenRepositoryAdapter.kt
package com.study.platform.global.auth.adapter

import com.study.platform.global.auth.port.TokenRepository
import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.stereotype.Component
import java.time.Duration

@Component
class RedisTokenRepositoryAdapter(
    private val stringRedisTemplate: StringRedisTemplate
) : TokenRepository {

    override fun save(key: String, value: String, ttl: Duration) {
        stringRedisTemplate.opsForValue().set(key, value, ttl)
    }

    override fun find(key: String): String? =
        stringRedisTemplate.opsForValue().get(key)

    override fun delete(key: String) {
        stringRedisTemplate.delete(key)
    }
}
